Kernel Status (#459)
executablebooks/thebe: docs/ui.rst

==============
User Interface
==============

Thebe includes support for some built-in user interface elements.

> The aim is to provide simple elements that can be added to a page a styled accordingly by the consumer without neeing to [re]write coont code.

Reviewer: ```suggestion
The aim is to provide simple elements that can be added to a page and styled accordingly by the consumer without needing to [re]write code.
```

Using these UI elements in the way shown here, assumes that you already have Thebe statically loaded on page.

Built in UI elements are:

- Activate Button
- Kernel Status Widget

Activate Button
===============

> Configure this UI element as follows.

Reviewer: ```suggestion
When this button is pressed, it will run the `thebelab.bootstrap` function on the page.
Configure this UI element as follows.
```

> Add a :code:`div` element to the page in the desired location.

Reviewer: ```suggestion
Add a ``div`` element to the page in the desired location.
```

I think double backticks does the same thing in rST (we should really convert this to MyST one day 😬 )

.. code-block:: html

  <div class="thebe-activate" />

Then set the following option to :code:`mountActivateWidget:true` in the Thebe config object

Status Widget
=============

> Configure this UI element as follows.

Reviewer: ```suggestion
This element will display the status of the kernel, it will update when the kernel sends new messages or when code is executed.
Configure this UI element as follows.
```

.. code-block:: html

  <script src="https://unpkg.com/thebe@latest/lib/index.js"></script>
  <script type="text/x-thebe-config">
    {
      requestKernel: true,
      binderOptions: {
        repo: "binder-examples/requirements",
      },
      mountActivateWidget: true,
      mountStatusWidget: true
    }
  </script>
  <div class="thebe-activate" />
  <div class="thebe-status" />

Pressing activate will convert the following code into an activate cell, and the kernel status widget
will show status of the kernel launch.

.. raw:: html

  <script src="https://unpkg.com/thebe@latest/lib/index.js"></script>
  <script type="text/x-thebe-config">
    {
      requestKernel: true,
      binderOptions: {
        repo: "binder-examples/requirements",
      },
      mountActivateWidget: true,
      mountStatusWidget: true
    }
  </script>
  <div class="thebe-activate"></div>
  <div class="thebe-status"></div>

.. raw:: html

   <pre data-executable="true" data-language="python">
   %matplotlib inline
   import numpy as np
   import matplotlib.pyplot as plt
   x = np.linspace(0,10)
   plt.plot(x, np.sin(x))
   plt.plot(x, np.cos(x))
   </pre>

> Note: If you are looking to load Thebe dynamically, check the custom launch button in the example `here. <https://github.com/executablebooks/thebe/blob/feat/kernel-status/examples/demo-launch-button.html>`_

Reviewer: ```suggestion
Note: If you are looking to load Thebe dynamically, check the custom launch button `in the example here. <https://github.com/executablebooks/thebe/blob/feat/kernel-status/examples/demo-launch-button.html>`_
```

I find that it's easier to linkify multiple words so people can see the link more easily

Author: i like it 👍
